### PR TITLE
Respect tabbedview settings when generating an dossier or task excel export.

### LINKED ISSUE #5469: Excel export should respect tabbedview-settings.
_Harvest: `OneGov GEVER` - `CR 2019.2 SG`_

The excel export that is available for listing via `Weitere Aktionen` -> `Auswahl exportieren` should also only include the columns that are currently selected in the users tabbed-view setting. Also the order of the columns as customized by the current user should be respected.

Care should be taken to make sure this works in all places where such an export is available.

https://extranet.4teamwork.ch/support/gever-st-gallen/tracker/627

### PR DESCRIPTION
**Pull request by phgross**
The personal tabbeview extjs listing settings (visible columns, column order) are now respected in Excel exports (closes #5469).

Implementation details and additional work:
 - To fetch the correct tabbedview setting i add the `view_name` as an additional tabbedview form value.
 - Fetching the currently visible columns and their order is implemented conservative, so that the default settings are used, when no personal settings exists or the settings couldn't be loaded.
 - Convert all existing tests to integration tests.
 - Implement base class for excel export views  and smaller refactorings.

### CODE REVIEW

File: opengever/dossier/browser/report.py [hunk 1]
```diff
@@ -16,8 +20,9 @@ class DossierReporter(BrowserView):
     """
 
     def get_dossier_attributes(self):
-        return [
+        attributes = [
```
Comment (lukasgraf) [inline opengever/dossier/browser/report.py]: These are rather columns than attributes, no? (Both in the Excel sheet as well as in the TabbedView table).
So I would probably rather go with `excel_columns` here...

File: opengever/dossier/browser/report.py [hunk 2]
```diff
@@ -37,6 +42,47 @@ def get_dossier_attributes(self):
                         default=u'Reference Number')},
         ]
 
+        return self.filter_and_order_by_tabbedview_settings(attributes)
+
+    def filter_and_order_by_tabbedview_settings(self, attributes):
```
Comment (lukasgraf) [inline opengever/dossier/browser/report.py]: `attributes` -> `excel_columns`

File: opengever/dossier/browser/report.py [hunk 3]
```diff
@@ -37,6 +42,47 @@ def get_dossier_attributes(self):
...
+        attributes_dict = {
```
Comment (lukasgraf) [inline opengever/dossier/browser/report.py]: `excel_columns_by_name`?

File: opengever/dossier/tests/test_reporter.py [hunk 4]
```diff
@@ -1,116 +1,109 @@
...
 
-class TestDossierReporter(FunctionalTestCase):
+    @browsing
+    def test_respects_column_tabbedview_settings_if_exists(self, browser):
+        self.login(self.regular_user, browser=browser)
 
-    def setUp(self):
-        super(TestDossierReporter, self).setUp()
+        data = {'view_name': 'mydossiers',
+                'gridstate': '{"columns":[{"id":"path_checkbox","width":30},{"id":"sortable_title","width":300,"sortable":true},{"id":"review_state","width":110,"sortable":true},{"id":"reference","width":110,"sortable":true},{"id":"end","width":110,"sortable":true},{"id":"responsible","width":110,"sortable":true},{"id":"start","width":110,"hidden":true,"sortable":true},{"id":"Subject","width":110,"sortable":true},{"id":"dummy","width":1,"hidden":true}],"sort":{"field":"modified","direction":"ASC"}}'}
```
Comment (lukasgraf) [inline opengever/dossier/tests/test_reporter.py]: I would also prefer the same style (using `json.dumps` etc..) here like the one you used for the task report tests.

**Comment by phgross** [comment]
@lukasgraf updated ...

**Pull request merged**